packages/server/src/arpa_reporter/lib/audit-report.js

```diff
@@ -360,7 +426,97 @@ function createHeadersProjectSummariesV2(projectSummaryGroupedByProject) {
     return headers;
 }
 
-async function generate(requestHost, tenantId) {
+async function runCache(domain, reportingPeriod = null, tenantId = null, periodId = null) {
+    if (reportingPeriod == null) {
+        const reportingPeriods = await getPreviousReportingPeriods(periodId);
+        const previousReportingPeriods = reportingPeriods.filter((p) => p.id !== periodId);
+        reportingPeriod = previousReportingPeriods
+            .reduce((a, b) => (a.id > b.id ? a : b));
+    }
+    const cacheFilename = cacheFSName(reportingPeriod, tenantId);
+    const data = await module.exports.generateSheets(reportingPeriod.id, domain, tenantId, null);
+    const jsonData = JSON.stringify(data);
+    await fs.mkdir(path.dirname(cacheFilename), { recursive: true });
+    await fs.writeFile(cacheFilename, jsonData, { flag: 'wx' });
+    return data;
+}
+
+function reviveDate(key, value) {
+    // Matches strings like "2022-08-25T09:39:19.288Z"
+    const isoDateRegex = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z$/;
+    return typeof value === 'string' && isoDateRegex.test(value)
+        ? new Date(value)
+        : value;
+}
+
+async function getCache(periodId, domain, tenantId = null, force = false, logger = log) {
```

Comment:
See comment above: Using `null` as a default value prevents any subsequent calls from falling back to their default parameters.